src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -618,6 +618,14 @@ private SingleNamespaceOrTypeDeclaration VisitTypeDeclaration(TypeDeclarationSyn
                 declFlags |= SingleTypeDeclaration.TypeDeclarationFlags.HasAnyNontypeMembers;
             }
 
+            if (node is RecordDeclarationSyntax recordDeclaration)
+            {
+                var messageId = recordDeclaration.Kind() == SyntaxKind.RecordStructDeclaration
```

Comment:
Should we also be checking for `record class`, as that wasn't in C# 9?

Reply:
that's what the falsy case is.  I could make it explicit and throw if neither of those cases if you prefer!

Reply:
I do not understand. I was expecting to see a check to see if `class` was explicitly specified by the user, and if so use `MessageID.IDS_FeatureRecordStructs` instead of `FeatureRecords`. I'm not sure what you mean by falsey here.

Reply:
oh, i think i see what you mean.  was it the case that 'record class' came about at the same time as 'record struct'?  I thought we have 'record/record-class' to begin with, and then added record-struct later.  If 'record class' is only legal in c# 10, then yes i'll have to fix this up.  good to know!

Reply:
done.